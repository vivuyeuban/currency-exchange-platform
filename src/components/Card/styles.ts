import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) => createStyles({
  container: {
    borderRadius: 20,
    boxShadow: theme.shadows[1],
    width: 350,
    height: 400,
    backgroundColor: `${theme.palette.background.paper}`,
    display: 'flex',
    position: 'relative',
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignItems: 'strech',
    padding: '20px',
    overflow: 'hidden',
  },
}));
