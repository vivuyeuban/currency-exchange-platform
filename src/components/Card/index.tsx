import React from 'react';
import { useStyles } from './styles';

const Card: React.FunctionComponent<{ children: React.ReactNode }> = React.memo(({
  children,
}) => {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      {children}
    </div>
  );
});

export default Card;
