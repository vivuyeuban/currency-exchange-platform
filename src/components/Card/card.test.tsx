import React from 'react';
import { render } from '@testing-library/react';
import Card from './index';

test('renders card', () => {
  const { container } = render(<Card />);
  expect(container).toMatchSnapshot();
});
