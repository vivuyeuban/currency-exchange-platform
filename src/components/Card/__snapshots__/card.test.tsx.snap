// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  />
</div>
`;
